packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -182,8 +182,16 @@ class AnalyzeCommand extends FlutterCommand {
           .where((String line) => !line.startsWith(new RegExp(r'^ *#')))
           .forEach((String line) {
             int colon = line.indexOf(':');
-            if (colon > 0)
-              dependencies.add(line.substring(0, colon), path.normalize(path.absolute(directory.path, path.fromUri(line.substring(colon+1)))), dotPackagesPath);
+            if (colon > 0) {
+              String packageName = line.substring(0, colon);
+              String packagePath = path.fromUri(line.substring(colon+1));
+              // Ensure that we only add the `analyzer` package defined in the vended SDK (and referred to with a local path directive).
+              // Analyzer package versions reached via transitive dependencies (e.g., via `test`) are ignored since they would produce
+              // spurious conflicts.
+              if (packageName != 'analyzer' || packagePath.startsWith('..')) {
```

Comment:
I patched this in and it became obvious that we need this change to get out of the `flutter update-packages --upgrade` conundrum, but I think we should limit this filter to only when we are analyzing the flutter repo itself.

I recommend changing
old: `if (packageName != 'analyzer' || ...`
new: `if (flutterRepo && (packageName != 'analyzer' || ...`

 I think that given our current use of the `analyzer` package API, this is an reasonable hack so that we don't have to spread the `dependency_override` around any wider than what's here in this PR. We **will** hit a snag if the `analyzer` API we use differs between the `analyzer` we specify in the flutter_tools `pubspec.yaml` and the `analyzer` API in the `bin\cache\dart-sdk`. Long term I still think that we should call `dartanalyzer` rather than consuming the `analyzer` package, but it's going to take some effort and time to get there and this is a great bandaid until then.